### ISSUE TITLE
[REVIEW]: home2park: An R package to assess the spatial provision of urban parks

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3609]
**Submitting author:** @xp-song (<a href="http://orcid.org/0000-0002-8825-195X">Xiao Ping Song</a>)
**Repository:** <a href="https://github.com/ecological-cities/home2park" target ="_blank">https://github.com/ecological-cities/home2park</a>
**Version:** v0.1.1
**Editor:** @martinfleis
**Reviewer:** @aelissa, @nickbearman
**Archive:** <a href="https://doi.org/10.5281/zenodo.5516698" target="_blank">10.5281/zenodo.5516698</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0/status.svg)](https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0"><img src="https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0/status.svg)](https://joss.theoj.org/papers/16d81fa40c903d4cd5dfa91900b55bb0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ecological-cities/home2park" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@aelissa & @nickbearman, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @martinfleis know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @aelissa
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ecological-cities/home2park">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ecological-cities/home2park">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@xp-song) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @nickbearman
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/ecological-cities/home2park">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ecological-cities/home2park">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@xp-song) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Wordcount for `paper.md` is 678

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2478/quageo-2019-0008 is OK
- 10.1016/j.apgeog.2015.06.008 is OK
- 10.1007/s10708-009-9303-8 is OK
- 10.1016/j.landurbplan.2016.11.001 is OK
- 10.1038/s41598-020-57864-4 is OK
- 10.3390/ijerph14121521 is OK
- 10.1016/j.landurbplan.2020.103908 is OK
- 10.1088/1748-9326/abb396 is OK
- 10.1016/j.ufug.2020.126689 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.21 s (423.5 files/s, 105467.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
HTML                            34           1805            484           6890
JavaScript                      12            797            543           3866
Markdown                         6            307              0           1077
CSS                              6            157             77           1045
R                               19            596           1601           1032
XML                              2              0              2            525
Rmd                              4            351            398            307
YAML                             4             15              2            142
TeX                              1              0              0            134
SVG                              1              0              1             11
-------------------------------------------------------------------------------
SUM:                            89           4028           3108          15029
-------------------------------------------------------------------------------


Statistical information for the repository 'ffd94db9aa3d6b4dddb864de' was
gathered on 2021/08/12.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
xpsong                           2          5242             24          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
xpsong                     5218           99.5          0.0               10.44
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: :point_left:

**Comment by martinfleis** [comment]
👋  @xp-song @aelissa @nickbearman this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#3609` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 2-4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me (@martinfleis) if you have any questions/concerns.

**Comment by nickbearman** [comment]
Hi all, I am having a look at this now. 
For the record, as it is a Saturday:
My working pattern is probably not the same as your working pattern, therefore
you may get (GitHub notifications) from me outside of normal working hours. Please do not feel any 
pressure to respond outside of your own working pattern.

**Comment by nickbearman** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by nickbearman** [comment]
@martinfleis I think my invitation has expired, I can't edit the checklist and when I click the link https://github.com/openjournals/joss-reviews/invitations GitHub said the invitation had expired. Thanks!

**Comment by martinfleis** [comment]
@whedon re-invite @nickbearman as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@nickbearman please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by martinfleis** [comment]
@nickbearman I have invited you again. Let me know if the issue persists. Thanks!

**Comment by nickbearman** [comment]
@martinfleis Thanks very much, it works now!

**Comment by nickbearman** [comment]
I have a query reinstallation. 
On my own PC I had to install Rtools (which was fine) and then reinstall a couple of specific packages (ellipsis and backports) but I think this is just the usual R package fiddling around with that we need to do. I did get the package working on my PC. 

To verify that it should do a clean install, I tried installing it on RStudio.Cloud. That, however, did not work:
`Installing package into ‘/cloud/lib/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
* installing *source* package ‘home2park’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
Killed
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpDVcFS8/filee043922ee/home2park_0.1.0.tar.gz’ had non-zero exit status`

Given this needs to be complied, does anyone know if that should be possible on RStudio.Cloud?
I have no other way of testing install on a 'clean' environment! Thanks

**Comment by nickbearman** [comment]
Currently I can't really test the software without a link to the 'Get started' tutorial. 
I have run through the README instructions, but they are quite high-level. 
The paper could expand slightly on the statistic the library generates - this is not 100% clear. Possibly a list of statistics it generates might be useful? I'm not sure currently, maybe the tutorial will help. I will revisit this in a few days time and see if it is any clearer.

**Comment by xp-song** [comment]
> Currently I can't really test the software without a link to the 'Get started' tutorial.
> I have run through the README instructions, but they are quite high-level.

Thanks for pointing out the dead link, I've amended it in commit [91e1f09](https://github.com/ecological-cities/home2park/commit/91e1f098e2671c2eb3103f96db023da6f9c3bf81). The website also contains more detailed explanations of the package: https://ecological-cities.github.io/home2park/

I'll have a look at testing the installation on other environments and see if it works, perhaps using [conda](https://stackoverflow.com/questions/45383754/simplest-way-to-test-installation-of-r-package-on-fresh-r-install)

**Comment by xp-song** [comment]
> I have a query reinstallation.
> On my own PC I had to install Rtools (which was fine) and then reinstall a couple of specific packages (ellipsis and backports) but I think this is just the usual R package fiddling around with that we need to do. I did get the package working on my PC.
> 
> To verify that it should do a clean install, I tried installing it on RStudio.Cloud. That, however, did not work:
> `Installing package into ‘/cloud/lib/x86_64-pc-linux-gnu-library/4.1’
> (as ‘lib’ is unspecified)
> 
> * installing _source_ package ‘home2park’ ...
>   ** using staged installation
>   ** R
>   ** data
>   *** moving datasets to lazyload DB
>   Killed
>   Warning message:
>   In i.p(...) :
>   installation of package ‘/tmp/RtmpDVcFS8/filee043922ee/home2park_0.1.0.tar.gz’ had non-zero exit status`
> 
> Given this needs to be complied, does anyone know if that should be possible on RStudio.Cloud?
> I have no other way of testing install on a 'clean' environment! Thanks

I'm not too experience with testing with 'clean' installation myself (other than the [R-CMD-Check](https://github.com/r-quantities/units/issues/223), which works fine), but based on [this post](https://stackoverflow.com/questions/45383754/simplest-way-to-test-installation-of-r-package-on-fresh-r-install), I managed to install it using a new conda environment:

**In my command line (Mac Terminal)**:
> `conda create -n r-dev -c r r-essentials`
> `conda activate r-dev`

To install packages from github, install R package `devtools`:
> `conda install r-devtools`


To ensure that dependency R package `units` [installs properly](https://github.com/r-quantities/units/issues/223), particularly for a conda environment:
> `conda install -c conda-forge udunits2` 


Load R, and run the following:
> `devtools::install_github("ecological-cities/home2park@main")`



---

`sessionInfo()`

> R version 3.6.1 (2019-07-05)
> Platform: x86_64-apple-darwin13.4.0 (64-bit)
> Running under: macOS  10.16
> 
> Matrix products: default
> BLAS/LAPACK: /Users/xpsong/opt/anaconda3/envs/r-dev/lib/R/lib/libRblas.dylib
> 
> locale:
> [1] C/UTF-8/C/C/C/C
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> loaded via a namespace (and not attached):
> [1] compiler_3.6.1

**Comment by nickbearman** [comment]
Thanks @xp-song for the notes re installation. I'm happy with this. @martinfleis any comments?

**Comment by nickbearman** [comment]
Thanks @xp-song also for the tutorial link. This really helps. 

On the tutorial itself, could you include the code you use to create the maps? I think this would be really useful for the tutorial. 

Also, the example data is a bit slow to run on my machine. Would it be possible to trim it down a bit, so the code runs faster?

**Comment by nickbearman** [comment]
Re the article, I have a conceptual question. You talk about
" the fact that supply from parks further away are reduced"
Which I think I understand (i.e. the closer people are to a park, the more likely they are to use it). 
Could you rephrase it so we talk about people using the park, rather than the park being supplied to the people? I think this would make the article much easier to follow. 

You also mention " By providing fine-grained measures for a variety of park attributes"
Can you list these attributes in the article? earlier on (2. Summarise at each park...) you mention attributes. Are these the same, or different?

**Comment by xp-song** [comment]
> Thanks @xp-song also for the tutorial link. This really helps.
> 
> On the tutorial itself, could you include the code you use to create the maps? I think this would be really useful for the tutorial.
> 

Added in commit [703c76f](https://github.com/ecological-cities/home2park/commit/703c76f0858bec4a2b1bf5db5f35aa1903c89b57)

> Also, the example data is a bit slow to run on my machine. Would it be possible to trim it down a bit, so the code runs faster?

Amended example building dataset in commit [e50da8c](https://github.com/ecological-cities/home2park/commit/e50da8c9c35acdf9f9ec9bc9569db8e23ced809c)

**Comment by xp-song** [comment]
Thanks @nickbearman for raising some good points. Some notes and responses:

> Re the article, I have a conceptual question. You talk about
> " the fact that supply from parks further away are reduced"
> Which I think I understand (i.e. the closer people are to a park, the more likely they are to use it).
> Could you rephrase it so we talk about people using the park, rather than the park being supplied to the people? I think this would make the article much easier to follow.

Research on park provision tends to differentiate between '_supply_' (provision which may or may not be used), '_demand_', as well as actual '_use_'. The fact that parks are near to people's homes may not necessarily mean that people are actively '_using_' these places (which may be affected by factors other than distance). For e.g., desirability, cleanliness (on the supply side), or people's demographics/preferences such as whether they have children or pets (on the demand side) all affect actual park '_use_'. This is why the article/package documentation focuses on park '_supply_'.

 
> You also mention " By providing fine-grained measures for a variety of park attributes"
> Can you list these attributes in the article? earlier on (2. Summarise at each park...) you mention attributes. Are these the same, or different?

I've amended the article text to try to clarify what I mean by "park attributes" in commit [55b1e77](https://github.com/ecological-cities/home2park/commit/55b1e777a7750800ccb9945ffa72c49e14976db9), and have also provided some examples. Thanks for helping to make the article clearer!

**Comment by xp-song** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: :point_left:

**Comment by martinfleis** [comment]
> Thanks @xp-song for the notes re installation. I'm happy with this. @martinfleis any comments?

If you're happy I am fine with that.

**Comment by aelissa** [comment]
Just started looking at this. 
Re the installation in the Vignette you suggest to run `devtools::install_github("ecological-cities/home2park")`. However,  while devtools::install_github uses master as default branch reference, your default is main. I would advise @xp-song to amend the installation command adding the correct ref `devtools::install_github("ecological-cities/home2park", ref="main")`

**Comment by xp-song** [comment]
> Just started looking at this.
> Re the installation in the Vignette you suggest to run `devtools::install_github("ecological-cities/home2park")`. However, while devtools::install_github uses master as default branch reference, your default is main. I would advise @xp-song to amend the installation command adding the correct ref `devtools::install_github("ecological-cities/home2park", ref="main")`

@aelissa Thanks for pointing this out, I've amended this is commit [8930b7c](https://github.com/ecological-cities/home2park/commit/8930b7cd080d396614af63e758b1eb86943d1712)

**Comment by aelissa** [comment]
I assessed the functionalities both by following the tutorial with the example data and by using some data for England. The tutorial is reproducible and the functions easily applicable to other data. Overall, I think it is a very useful package for analyzing parks' provision. @xp-song  just a few comments:

Re Functionality 

on rasterise_pop it seems that the population column needs to be named "pop count" to work. Try: 
``pop_sgp<-dplyr::rename(pop_sgp,pop=pop_count)``

``pop_rasters <- rasterise_pop(pop_sgp,
                           census_block = "subzone_n",
                           pop_count = "pop")``

Re Documentation

In the documentation of get_building_osm you might want to specify the date's format "%y%m%d"

There is a typo in rasterise_pop documentation at Description "such that the the integrated density"

In pop_dasymap and the rasterising functions I think you should clarify a bit more how the output data structure includes time. My understanding it that you have one layer for each year. This needs to be made more explicit in the rasterising functions documentations as well as in pop_dasymap -- if you do not rely on external raster but on those created with the pkg functions -- it should be made clearer that you need to pick the element of a list of rasters related to the year of interest.

I will look at the software paper and finalize my review in the next couple of days tops @martinfleis

**Comment by xp-song** [comment]
> I assessed the functionalities both by following the tutorial with the example data and by using some data for England. The tutorial is reproducible and the functions easily applicable to other data. Overall, I think it is a very useful package for analyzing parks' provision. @xp-song just a few comments:
> 
> Re Functionality
> 
> on rasterise_pop it seems that the population column needs to be named "pop count" to work. Try:
> `pop_sgp<-dplyr::rename(pop_sgp,pop=pop_count)`
> 
> `pop_rasters <- rasterise_pop(pop_sgp, census_block = "subzone_n", pop_count = "pop")`
> 
> Re Documentation
> 
> In the documentation of get_building_osm you might want to specify the date's format "%y%m%d"
> 
> There is a typo in rasterise_pop documentation at Description "such that the the integrated density"
> 
> In pop_dasymap and the rasterising functions I think you should clarify a bit more how the output data structure includes time. My understanding it that you have one layer for each year. This needs to be made more explicit in the rasterising functions documentations as well as in pop_dasymap -- if you do not rely on external raster but on those created with the pkg functions -- it should be made clearer that you need to pick the element of a list of rasters related to the year of interest.
> 
> I will look at the software paper and finalize my review in the next couple of days tops @martinfleis

Thank you! I've made the respective amendments in the linked commits above

**Comment by aelissa** [comment]
Brilliant. I have just looked at the software paper too and I am happy with it @martinfleis

**Comment by martinfleis** [comment]
Thank you @aelissa for your review and @xp-song for a quick reaction!

@nickbearman are there any outstanding issues on your side?

**Comment by nickbearman** [comment]
> Thanks @nickbearman for raising some good points. Some notes and responses:
> 
> > Re the article, I have a conceptual question. You talk about
> > " the fact that supply from parks further away are reduced"
> > Which I think I understand (i.e. the closer people are to a park, the more likely they are to use it).
> > Could you rephrase it so we talk about people using the park, rather than the park being supplied to the people? I think this would make the article much easier to follow.
> 
> Research on park provision tends to differentiate between '_supply_' (provision which may or may not be used), '_demand_', as well as actual '_use_'. The fact that parks are near to people's homes may not necessarily mean that people are actively '_using_' these places (which may be affected by factors other than distance). For e.g., desirability, cleanliness (on the supply side), or people's demographics/preferences such as whether they have children or pets (on the demand side) all affect actual park '_use_'. This is why the article/package documentation focuses on park '_supply_'.
> 
> > You also mention " By providing fine-grained measures for a variety of park attributes"
> > Can you list these attributes in the article? earlier on (2. Summarise at each park...) you mention attributes. Are these the same, or different?
> 
> I've amended the article text to try to clarify what I mean by "park attributes" in commit [55b1e77](https://github.com/ecological-cities/home2park/commit/55b1e777a7750800ccb9945ffa72c49e14976db9), and have also provided some examples. Thanks for helping to make the article clearer!

Thanks very much @xp-song this really does make things clearer

**Comment by nickbearman** [comment]
> > Thanks @xp-song also for the tutorial link. This really helps.
> > On the tutorial itself, could you include the code you use to create the maps? I think this would be really useful for the tutorial.
> 
> Added in commit [703c76f](https://github.com/ecological-cities/home2park/commit/703c76f0858bec4a2b1bf5db5f35aa1903c89b57)
> 
> > Also, the example data is a bit slow to run on my machine. Would it be possible to trim it down a bit, so the code runs faster?
> 
> Amended example building dataset in commit [e50da8c](https://github.com/ecological-cities/home2park/commit/e50da8c9c35acdf9f9ec9bc9569db8e23ced809c)

Thanks - these look good too

**Comment by nickbearman** [comment]
> Thank you @aelissa for your review and @xp-song for a quick reaction!
> 
> @nickbearman are there any outstanding issues on your side?

Thanks @martinfleis I am now happy with the paper.

**Comment by martinfleis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: :point_left:

**Comment by martinfleis** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2478/quageo-2019-0008 is OK
- 10.1016/j.apgeog.2015.06.008 is OK
- 10.1007/s10708-009-9303-8 is OK
- 10.1016/j.landurbplan.2016.11.001 is OK
- 10.1038/s41598-020-57864-4 is OK
- 10.3390/ijerph14121521 is OK
- 10.1016/j.landurbplan.2020.103908 is OK
- 10.1088/1748-9326/abb396 is OK
- 10.1016/j.ufug.2020.126689 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by martinfleis** [comment]
Thank you @nickbearman!

@xp-song The submission is now almost ready to be published. 

The next steps you need to do now:

- [ ] Make a new tagged release of the software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service
- [ ] Check that the archival deposit (e.g., in Zenodo) has the correct metadata. In particular, the title and author list should be identical to those of the paper.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by xp-song** [comment]
@nickbearman @aelissa @martinfleis Thank you very much for your thoughtful review and comments. Here are requested information:

> * [x]  Make a new tagged release of the software, and list the version tag of the archived version here.

https://github.com/ecological-cities/home2park/releases/tag/v0.1.1

> * [x]  Archive the reviewed software in Zenodo or a similar service

https://zenodo.org/record/5516698#.YUfmNi0RppQ

> * [x ]  Please list the DOI of the archived version here.

10.5281/zenodo.5516698

**Comment by martinfleis** [comment]
@whedon set 10.5281/zenodo.5516698 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5516698" target="_blank">10.5281/zenodo.5516698</a> is the archive.

**Comment by martinfleis** [comment]
@whedon set v0.1.1 as version

**Comment by whedon** [comment]
OK. v0.1.1 is the version.

**Comment by martinfleis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03609/joss.03609/10.21105.joss.03609.pdf) :page_facing_up: :point_left:

**Comment by martinfleis** [comment]
Thank you @xp-song. I'm going to hand this over now to the associate EiC on rotation for the final steps.

**Comment by martinfleis** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2478/quageo-2019-0008 is OK
- 10.1016/j.apgeog.2015.06.008 is OK
- 10.1007/s10708-009-9303-8 is OK
- 10.1016/j.landurbplan.2016.11.001 is OK
- 10.1038/s41598-020-57864-4 is OK
- 10.3390/ijerph14121521 is OK
- 10.1016/j.landurbplan.2020.103908 is OK
- 10.1088/1748-9326/abb396 is OK
- 10.1016/j.ufug.2020.126689 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2594

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2594, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by xp-song** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @xp-song, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by xp-song** [comment]
whoops, embarrassing mistake.

**Comment by kthyng** [comment]
@xp-song Don't worry, it happens all the time! It doesn't say who should input that command, after all!

**Comment by kthyng** [comment]
I see your zenodo archive is made and in place, your version is up to date, and your paper looks great! We can proceed!

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1439977274816835586) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2595
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03609](https://doi.org/10.21105/joss.03609)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by martinfleis** [comment]
Congratulations @xp-song!

Thank you again @aelissa and @nickbearman for you valuable reviews!

**Comment by kthyng** [comment]
Thanks all, especially for your reviews @aelissa and @nickbearman and for being so thorough as handling editor @martinfleis!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03609/status.svg)](https://doi.org/10.21105/joss.03609)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03609">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03609/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03609/status.svg
   :target: https://doi.org/10.21105/joss.03609
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03609/status.svg)](https://doi.org/10.21105/joss.03609)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
